res/controllers/Reloop-Mixage.scripts.js

```diff
@@ -0,0 +1,784 @@
+    "pfl": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+    "loop_enabled": {"function": function(_v, g) { Mixage.toggleReloopLED(g); }},
+    "loop_in": {"function": function(v, g) { if (v === 1) { Mixage.toggleLoopLED(g); } }},
+    "loop_out": {"function": function(v, g) { if (v === 1) { Mixage.toggleLoopLED(g); } }},
+    "sync_enabled": {"function": function(v, g, c) { Mixage.toggleLED(v, g, c); }},
+    "eject": {"function": function(_v, g) { Mixage.eject(g); }},
+};
+
+// ----- Internal variables functions -----
+
+// Set or remove functions to call when the state of a mixxx control changes
+Mixage.connectControlsToFunctions = function(group, remove) {
+    for (const control in Mixage.connectionMap) {
+        if (remove !== undefined) {
+            Mixage.connectionMap[control][group].disconnect();
+        } else {
+            Mixage.connectionMap[control][group] = engine.makeConnection(group, control, Mixage.connectionMap[control].function);
+        }
+    }
+};
+
+Mixage.init = function(_id, _debugging) {
+
+    // all button LEDs off
+    for (let i = 0; i < 255; i++) {
+        midi.sendShortMsg(0x90, i, 0);
+    }
+
+    // find controls and make engine connections for each channel in Mixage.channels
+    // A predefined list with channels is used instead of a for loop to prevent engine connections to be overwritten
+    Mixage.channels.forEach(function(channel) {
+        const deck = script.deckFromGroup(channel);
+        Mixage.connectControlsToFunctions(channel);
+
+        // set soft takeovers for effectslot amount
+        for (let effectSlot = 1; effectSlot <= Mixage.numEffectSlots; effectSlot++) {
+            const groupString = "[EffectRack1_EffectUnit"+ deck +"_Effect" + effectSlot + "]";
+            engine.softTakeover(groupString, "meta", true);
+        }
+
+        for (let effectUnit = 1; effectUnit <= Mixage.numEffectUnits; effectUnit++) {
+            // make connections for the fx on LEDs
+            const fxGroup = "group_"+channel+"_enable";
+            Mixage.fxOnConnection.push(engine.makeConnection("[EffectRack1_EffectUnit"+effectUnit+"]", fxGroup, function() { Mixage.toggleFxLED(channel); }));
+
+            // set soft takeovers for effectunit meta
+            engine.softTakeover("[EffectRack1_EffectUnit"+effectUnit+"]", "super1", true);
+            engine.setValue("[EffectRack1_EffectUnit"+effectUnit+"]", "show_focus", 1);
+        }
+
+        // set soft takeover for filter effect
+        engine.softTakeover("[QuickEffectRack1_"+channel+"]", "super1", true);
+
+        // make connections for status LEDs
+        Mixage.vuMeterConnection.push(engine.makeConnection(channel, "vu_meter", function(val) { midi.sendShortMsg(0x90, Mixage.ledMap[channel].vu_meter, val * 7); }));
+        Mixage.loopConnection.push(engine.makeConnection(channel, "track_loaded", function() { Mixage.toggleReloopLED(channel); }));
```

Comment:
what led is this controlling?

Reply:
It's controlling the Reloop LED. when a new track is loaded that has a loop point set it starts the slowly blinking the reloop led (1s on 1s off). When a loop has not been set and thus reloop is not available it disables the blinking

Reply:
Ah yeah, should've read the mapping more carefully.